osgbuild/fetch_sources.py

```diff
@@ -330,14 +364,14 @@ def fancy_source_error(meta_type, explicit_type, handler, args, kw, e):
 
     xtype = "type" if explicit_type else "implicit type"
     log.error("Error processing source line of %s '%s'" % (xtype, meta_type))
-    varnames = handler.__code__.co_varnames
-    fn_argcount = handler.__code__.co_argcount
-    minargs = fn_argcount - len(handler.__defaults__)
-    reqargs = varnames[:minargs]
-    maxargs = varnames.index('ops')
-    posargs = varnames[:maxargs]
-    posargs_provided = posargs[:len(args)]
-    dupe_args = set(posargs_provided) & set(kw)
+    varnames = handler.__code__.co_varnames     # arg names for handler func
+    fn_argcount = handler.__code__.co_argcount  # number of positional args
+    minargs = fn_argcount - len(handler.__defaults__)  # number of required args
+    reqargs = varnames[:minargs]     # names of required arguments
+    maxargs = varnames.index('ops')  # max number of positional args we allow
```

Comment:
So this works because `ops` is the last argument in each of the functions?

Reply:
Yes, as is required by all fetch handler functions.

See:
https://github.com/opensciencegrid/osg-build/blob/v1.15.1/osgbuild/fetch_sources.py#L93-L101

Reply:
It goes after the last allowed positional argument, that is.  It's a way of marking that the remaining arguments are not allowed to be specified as positional arguments, but must explicitly be specified as `key=val` pairs.